### ISSUE TITLE
Dreamcast MMU/WinCE Support

### ISSUE THREAD

**Issue by GoogleCodeExporter** [aloone/nulldc#61]
```
i hope in future,nulldc can support dreamcast vmu,support windows ce games
```

Original issue reported on code.google.com by `tq03q9i@gmail.com` on 19 May 2010 at 7:35

**Comment by GoogleCodeExporter** [comment]
```
Very Sorry,MMU support,I'm Spelling mistake
```

Original comment by `tq03q9i@gmail.com` on 19 May 2010 at 7:38

**Comment by GoogleCodeExporter** [comment]
```
Better? :)
```

Original comment by `KrossX3` on 26 May 2010 at 4:06
- Changed title: **Dreamcast MMU Support**
- Added labels: **Type-Enhancement**
- Removed labels: **Type-Defect**

**Comment by GoogleCodeExporter** [comment]
```
yes,because i hope nudlldc runing super robot wars alpha for dreamcast,that is 
a 
great game
```

Original comment by `tq03q9i@gmail.com` on 26 May 2010 at 1:11

**Comment by GoogleCodeExporter** [comment]
```
Let's keep this one as a placeholder for related issues.
```

Original comment by `KrossX3` on 31 May 2010 at 3:49
- Changed title: **Dreamcast MMU/WinCE Support**
- Added labels: **Priority-Low**
- Removed labels: **Priority-Medium**

**Comment by GoogleCodeExporter** [comment]
```
Issue 11 has been merged into this issue.
```

Original comment by `KrossX3` on 31 May 2010 at 3:49

**Comment by GoogleCodeExporter** [comment]
```
Issue 75 has been merged into this issue.
```

Original comment by `KrossX3` on 31 May 2010 at 3:49

**Comment by GoogleCodeExporter** [comment]
```
Issue 101 has been merged into this issue.
```

Original comment by `Mr.PsyMan` on 31 May 2010 at 5:18

**Comment by GoogleCodeExporter** [comment]
```
Support!
```

Original comment by `rompecal...@gmail.com` on 14 Jun 2010 at 1:12

**Comment by GoogleCodeExporter** [comment]
```
Issue 152 has been merged into this issue.
```

Original comment by `KrossX3` on 26 Jun 2010 at 8:17

**Comment by GoogleCodeExporter** [comment]
```
rompecalson, are you saying that there is support for MMU or support for WinCE? 
 I thought MMU was already supported in older versions (there was an extra 
executable with "mmu" a the end of the name).  Either way, I haven't been able 
to run any WinCE games still.
```

Original comment by `Joseph...@gmail.com` on 21 Jul 2010 at 10:15

**Comment by GoogleCodeExporter** [comment]
```
I also would like to see support for WinCE games. Armada is great fun and I'd 
love to play it again on my PC.
```

Original comment by `brandon....@gmail.com` on 27 Jul 2010 at 11:59

**Comment by GoogleCodeExporter** [comment]
```
Issue 179 has been merged into this issue.
```

Original comment by `KrossX3` on 31 Jul 2010 at 11:52

**Comment by GoogleCodeExporter** [comment]
```
Issue 182 has been merged into this issue.
```

Original comment by `KrossX3` on 4 Aug 2010 at 1:20

**Comment by GoogleCodeExporter** [comment]
```
Ah, WinCe. I know many good games for WinCe: Armada, Spider Man, Who Wants To 
Beat Up a Millionaire, Nightmare Creatures 2 etc. I am suprised that NullDC 
still have not WinCE support.
P.S. Spider Man (WinCE game) is playable on NullDC. It's strange :O
```

Original comment by `proletar...@gmail.com` on 3 Oct 2010 at 5:25

**Comment by GoogleCodeExporter** [comment]
```
I don't think Spider-Man's a Windows CE game, it doesn't show the WinCE logo on 
the 'Presented by Sega' boot screen.

There's lots of good WinCE games which would be nice to see on nullDC, 
StarLancer and Sega Rally 2 to add to those you listed.
```

Original comment by `matbur...@gmail.com` on 3 Oct 2010 at 7:54

**Comment by GoogleCodeExporter** [comment]
```
The question is, is it worth a 10-15% slowdown + a few weeks of development it 
requires ?
```

Original comment by `drkIIRaziel` on 4 Oct 2010 at 8:42

**Comment by GoogleCodeExporter** [comment]
```
Ehm... yes!?
```

Original comment by `Mr.PsyMan` on 4 Oct 2010 at 9:06

**Comment by GoogleCodeExporter** [comment]
```
@drkIIRaziel:  I think it's worth the 10-15% slowdown for those of us with 
computers already more than capable of handling most, if not all, games at full 
speed.  Maybe have an option to disable WinCE support for users with slower 
computers or who just don't want it enabled for non-WinCE games.

Regarding the few weeks of development required to implement it, that is up to 
you and any other coders who would be working on it.
```

Original comment by `Joseph...@gmail.com` on 4 Oct 2010 at 9:16

**Comment by GoogleCodeExporter** [comment]
```
Would the 10-15% slowdown have to apply at all times, or just for WinCE games? 
Could a WinCE game be auto-detected (by opcode maybe) and then the extra code 
kicks in?

And yes, we'd love you forever raz if you add WinCE support. I might even 
donate :O
```

Original comment by `matbur...@gmail.com` on 4 Oct 2010 at 10:33

**Comment by GoogleCodeExporter** [comment]
```
Well, it'd make the dynarec much more complex, and i don't want to dupplicate 
code all over, so it'l affect speed on all games. Not as much as it'l affect 
wince ones though.

 JosephRAA: My timeslice for nulldc is allocated for all the ndc related deving. By doing mmu i will not do something else :p
```

Original comment by `drkIIRaziel` on 4 Oct 2010 at 12:30
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
I wonder if I'll be and emulation grandpa when raz will be done with MMU 
implementation...
```

Original comment by `Mr.PsyMan` on 4 Oct 2010 at 1:57

**Comment by GoogleCodeExporter** [comment]
```
Hopefully by then we'l have enough donations to get you a newer pc, so you can 
like .. test again ? Seriously, beeing tied to dx9 is no half fun -.-
```

Original comment by `drkIIRaziel` on 4 Oct 2010 at 2:15

**Comment by GoogleCodeExporter** [comment]
```
In our dreams.
```

Original comment by `Mr.PsyMan` on 4 Oct 2010 at 2:34

**Comment by GoogleCodeExporter** [comment]
```
WinCE support is the only feature missing in nullDC imo :) You should do it, 
guys!
```

Original comment by `rimmerp...@gmail.com` on 13 Oct 2010 at 12:22

**Comment by GoogleCodeExporter** [comment]
```
Really? So you think Dreamcast can't use microphone, right? What about NetPlay? 
What about VS Link Cable game? What about Dreameye (web cam for DC)? Dreamcast 
has so many features are not supported by ANY existing emulator (neither 
NullDC, nor Makaron or Demul). But they are not so important as WinCE support.
```

Original comment by `proletar...@gmail.com` on 17 Oct 2010 at 1:54

**Comment by GoogleCodeExporter** [comment]
```
He meant the only worthwhile feature ;)
```

Original comment by `matbur...@gmail.com` on 17 Oct 2010 at 5:15

**Comment by GoogleCodeExporter** [comment]
```
WinCE, after all, very restricted, limited patform (when it comes to 3D/Games 
capabilities), and DX6 can't be helped. So, games on WinCE has not-so-good 
graphics and, in most case, somewhat boring gameplay, so I think this priority 
is low deservedly. Also in general, they require mouse and keyboard for descent 
control, or real gamepad...
```

Original comment by `old.liq...@gmail.com` on 22 Dec 2010 at 2:34

**Comment by GoogleCodeExporter** [comment]
```
I do not think that Resident Evil 2 is boring or require mouse and keyboard ...
```

Original comment by `tapcio` on 22 Dec 2010 at 6:38

**Comment by GoogleCodeExporter** [comment]
```
I'm crying for WinCE support. The only DC emu supporting it is Makaron but it 
has some stability and gfx issues that makes games like Super Robot Wars Alpha 
unplayable. So I'm crossing my fingers for nullDC with WinCE support! :)
```

Original comment by `jacek.tr...@gmail.com` on 7 Jan 2011 at 10:56

**Comment by GoogleCodeExporter** [comment]
```
Please raise the priority on this one. There are so many great games that use 
WinCE. Here is a list of USA based ones:
4x4 Evolution
Armada
Atari Anniversary Edition
Bust-A-Move 4
Caesars Palace 2000
Championship Pro Surfer
Ducati World
Hidden and Dangerous
KISS Psycho Circus
Maximum Pool
Midway's Greatest Arcade Hits Vol. 1
Midway's Greatest Arcade Hits Vol. 2
Next Tetris, The - Online Edition
NFL Quarterback Club 2000
NFL Quarterback Club 2001
Nightmare Creatures II
Railroad Tycoon II
Rainbow Six
Resident Evil 2
Sega Rally 2
South Park: Chefs Luv Shack
Spirit of Speed 1937
Starlancer
Super Runabout: San Francisco Edition
Tomb Raider Chronicles
Tomb Raider The Last Revelation
Who Wants To Beat Up A Millionaire
Wild Metal
Worms Armageddon
Worms World Party

I have personally tested them and they all freeze a the Sega load screen give 
an OPCODE error

```

Original comment by `thelucid...@aol.com` on 30 Apr 2011 at 6:50

**Comment by GoogleCodeExporter** [comment]
```
wince is a portable os developed for mini-laptops and pda's so instead of 
begging for it to be emulated here why not try to run it in virtual box.
```

Original comment by `psion...@gmail.com` on 3 May 2011 at 5:34

**Comment by GoogleCodeExporter** [comment]
```
@psion...@gmail.com,

Because is not possible, VirtualBox is x86/x86_64 emulator, DC have SH-4 
processor. It's a complete different architecture.
```

Original comment by `tapcio` on 3 May 2011 at 9:52

**Comment by GoogleCodeExporter** [comment]
```
This is the single most significant outstanding issue w/ nullDC.

Aside from pathological and impractical speed issues (i.e., cases where *no* 
commodity hardware is fast enough), no optimizations in speed are a substitute 
for enormous gaps in functionality.

tl;dr: nobody cares how fast it is if it can't do what you want.
```

Original comment by `xalto...@gmail.com` on 19 Sep 2011 at 9:42

**Comment by GoogleCodeExporter** [comment]
```
I want Windows CE support in NullDC too :(
I wanna be able to play Super Robot Taisen and Sega Rally 2 again :/
```

Original comment by `omega...@gmail.com` on 21 Oct 2011 at 6:10

**Comment by GoogleCodeExporter** [comment]
```
You have my vote for Wince support! :)
```

Original comment by `OldEC...@gmail.com` on 16 Nov 2011 at 3:33

**Comment by GoogleCodeExporter** [comment]
```
WinCE requires creating a new emulator. If it will be created some day, I think 
this programm should be named like "NullDC-CE" or something... For example, now 
we have two emulators: NullDC & NullDC_NAOMI.

```

Original comment by `N2008eo...@rambler.ru` on 16 Nov 2011 at 8:30

**Comment by GoogleCodeExporter** [comment]
```
Demul supports winCE
```

Original comment by `sh.rahma...@gmail.com` on 8 Aug 2012 at 1:29

**Comment by GoogleCodeExporter** [comment]
```
Demul runs extremely slowly, not worth it.
```

Original comment by `BBoy.Vec...@gmail.com` on 27 Feb 2015 at 12:57
